plugins/ptp_operator/metrics/metrics.go

```diff
@@ -0,0 +1,381 @@
+			m.Stats[iface] = s
+			m.lock.Unlock()
+		}
+		s.frequencyAdjustment = frequencyAdjustment
+		s.delayFromMaster = delayFromMaster
+		s.lastOffset = offsetFromMaster
+
+		s.addValue(offsetFromMaster)
+		lastLockState := s.lastClockState
+		s.lastClockState = clockState
+		if phc2sysProcessName == processName {
+			GenPhc2SysEvent(iface, offsetFromMaster, clockState, lastLockState)
+			UpdatePTPMetrics(processName, iface, offsetFromMaster, s.getMaxAbs(), frequencyAdjustment, delayFromMaster)
+		} else if ptp4lProcessName == processName {
+			UpdatePTPMetrics(processName, iface, offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster)
+		}
+	} else if ptp4lProcessName == processName {
+		clockState := extractPTP4lEventState(processName, output)
+		if clockState == "" {
+			return
+		}
+		var s *Stats
+		var found bool
+		if s, found = m.Stats[iface]; !found {
+			m.lock.Lock()
+			s = &Stats{}
+			m.Stats[iface] = s
+			m.lock.Unlock()
+		}
+
+		lastLockState := s.lastClockState
+		s.lastClockState = clockState
+		if clockState != lastLockState {
+			log.Infof("EVENT DATA FOR %s  IS %#v", iface, eventData(clockState))
+		}
+	}
+}
+
+var registerMetrics sync.Once
+
+// RegisterMetrics ... register metrics for all side car plugins
+func RegisterMetrics(nodeName string) {
+	registerMetrics.Do(func() {
+		prometheus.MustRegister(OffsetFromMaster)
+		prometheus.MustRegister(MaxOffsetFromMaster)
+		prometheus.MustRegister(FrequencyAdjustment)
+		prometheus.MustRegister(DelayFromMaster)
+
+		// Including these stats kills performance when Prometheus polls with multiple targets
+		prometheus.Unregister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+		prometheus.Unregister(prometheus.NewGoCollector())
+
+		NodeName = nodeName
+	})
+}
+
+// UpdatePTPMetrics ...
+func UpdatePTPMetrics(process, iface string, offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster float64) {
+	OffsetFromMaster.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(offsetFromMaster)
+
+	MaxOffsetFromMaster.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(maxOffsetFromMaster)
+
+	FrequencyAdjustment.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(frequencyAdjustment)
+
+	DelayFromMaster.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(delayFromMaster)
+}
+
+func extractSummaryMetrics(processName, output string) (offsetFromMaster, maxOffsetFromMaster, frequencyAdjustment, delayFromMaster float64) {
+	// remove everything before the rms string
+	// This makes the out to equals
+	indx := strings.Index(output, "rms")
+	output = output[indx:]
+	fields := strings.Fields(output)
+
+	if len(fields) < 5 {
+		glog.Errorf("%s failed to parse output %s: unexpected number of fields", processName, output)
+		return
+	}
+
+	offsetFromMaster, err := strconv.ParseFloat(fields[1], 64)
```

Comment:
good idea, but that will not make it readable, since we are splitting the strings. I will leave it at index so it is easy to see the positioning from the string